packages/@aws-cdk-containers/ecs-service-extensions/lib/service.ts

```diff
@@ -283,28 +289,39 @@ export class Service extends Construct {
       throw new Error(`Unknown capacity type for service ${this.id}`);
     }
 
+    const scaleOnCpuExtension = this.serviceDescription.get('scale-on-cpu-utilization') as ScaleOnCpuUtilization;
+
+    if (props.autoScaleTaskCount && scaleOnCpuExtension) {
+      throw Error('Cannot specify \'minTaskCount\' and \'maxTaskCount\' both in the Service construct and the \'ScaleOnCpuUtilization\' extension (deprecated). Please only provide the values in either one.');
+    }
+
     // Create the auto scaling target and configure target tracking policies after the service is created
-    if (props.autoScaleTaskCount) {
+    const maxCapacity = props.autoScaleTaskCount?.maxTaskCount ?? scaleOnCpuExtension?.maxTaskCount;
+    if (maxCapacity) {
```

Comment:
If scaleOnCpuExtension.minTaskCount is defined, and scaleOnCpuExtension.maxTaskCount is undefined, then this will not create this.scalableTaskCount. Is this intended? I think that would probably be a regression from the logic that was previously in ScaleOnCpuExtension.useService().